packages/nvidia_nat_core/src/nat/front_ends/fastapi/message_handler.py

```
@@ -363,8 +374,14 @@ async def human_interaction_callback(self, prompt: InteractionPrompt) -> HumanRe
                 return HumanResponseNotification()
 
-            # Wait for the human response future to complete
-            text_content: TextContent = await human_response_future
+            backend_timeout_in_seconds: int | None = (prompt.content.timeout + self._HITL_TIMEOUT_GRACE_PERIOD_SECONDS
+                                                      if prompt.content.timeout is not None else None)
+            try:
+                text_content: TextContent = await asyncio.wait_for(human_response_future,
+                                                                   timeout=backend_timeout_in_seconds)
+            except TimeoutError:
+                raise TimeoutError(
+                    f"HITL prompt timed out after {prompt.content.timeout}s waiting for human response") from None
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)